test/selectors/dashboard-selectors_spec.js

```diff
@@ -0,0 +1,82 @@
+import { expect } from 'chai'
+import { describe, it } from 'mocha'
+import { fromJS } from 'immutable'
+import { getSortedStudents, getStudentProgress } from '../../js/selectors/dashboard-selectors'
+
+describe('dashboard selectors', () => {
+  const s1 = { id: 1, firstName: 'X', lastName: 'C' }
+  const s2 = { id: 2, firstName: 'y', lastName: 'b' }
+  const s3 = { id: 3, firstName: 'Z', lastName: 'A' }
+
+  describe('getStudentProgress', () => {
+    const state = fromJS({
+      report: {
+        students: { 1: s1, 2: s2, 3: s3 },
+        activities: {
+          1: { children: [ 1 ] },
+          2: { children: [ 2 ] }
+        },
+        sections: {
+          1: { children: [ 1 ] },
+          2: { children: [ 2 ] }
+        },
+        pages: {
+          1: { children: [ 'Q1' ] },
+          2: { children: [ 'Q2', 'Q3' ] }
+        },
+        questions: {
+          Q1: { answers: [ 'A1', 'A2' ] }, // activity 1
+          Q2: { answers: [ 'A3', 'A4' ] }, // activity 2
+          Q3: { answers: [ 'A5', 'A6' ] } // activity 2
+        },
+        answers: {
+          A1: { studentId: 1, type: 'SomeAnswer', submitted: true },
+          A2: { studentId: 2, type: 'NoAnswer' },
+          A3: { studentId: 1, type: 'SomeAnswer', submitted: false },
+          A4: { studentId: 2, type: 'SomeAnswer', submitted: true },
+          A5: { studentId: 1, type: 'SomeAnswer', submitted: true },
+          A6: { studentId: 2, type: 'NoAnswer' }
+        }
+      }
+    })
+
+    it('should return hash with student progress', () => {
+      expect(getStudentProgress(state).toJS()).to.eql(
+        {
+          1: {
+            1: 1, // activity 1
+            2: 0.5 // activity 2 - only one submitted answer
+          },
+          2: {
+            1: 0, // activity 1
+            2: 0.5 // activity 2 - one submitted answer
+          },
+          3: { // this student hasn't started any activity, no answer objects
+            1: 0,
+            2: 0
+          }
+        }
+      )
+    })
+  })
+
+  describe('getSortedStudents', () => {
+    describe('when sortBy = "name"', () => {
+      const state = fromJS({
+        report: {
+          students: { 1: s1, 2: s2, 3: s3 }
+        },
+        dashboard: {
+          sortBy: 'name'
+        }
+      })
+
+      it('should return sorted list of students', () => {
+        expect(getSortedStudents(state).toJS()).to.eql(
+          // Students sorted by name (last name first)
+          [ s3, s2, s1 ]
```

Comment:
I would add a test for first name sorting here. Looking at the selector, there's actually a bug where these students aren't sorted correctly:

```
const s1 = { id: 2, firstName: 'Y', lastName: 'AB' }
const s2 = { id: 3, firstName: 'Z', lastName: 'A' }
```

`s2` should come first, but `s1` comes first because the names are concatenated.